constellations/gemini-admin/src/achieve/achieveEdit.js

```diff
@@ -146,15 +147,34 @@ export class AchieveEditPage extends React.Component {
             <div id="view-achieve-edit">
                 {modalDiv}
                 <h2>{title}</h2>
-                <h4>Year</h4>
-                <input
+                <InputText
+                    label="Year"
+                    description="Enter the achievement year"
+                    required={true}
                     value={this.state.inputYear}
-                    onChange={(e) => this.handleChange(e, "inputYear")}
+                    onChangeCallback={(e) => this.handleChange(e, "inputYear")}
+                    validators={[
+                        {
+                            validate: (text) => parseInt(text) > 2000,
+                            message: "You must input a year greater than 2000",
+                        },
+                    ]}
                 />
-                <h4>Message</h4>
-                <input
+                <InputText
+                    label="Message"
+                    description="Enter the achievement message"
+                    isTextBox={true}
+                    required={true}
                     value={this.state.inputMessage}
-                    onChange={(e) => this.handleChange(e, "inputMessage")}
+                    onChangeCallback={(e) =>
+                        this.handleChange(e, "inputMessage")
+                    }
+                    validators={[
```

Comment:
There is an `emptyValidator` function in `InputText.js`.

Reply:
Look around for an example Tony has made before related to `emptyValidator`. If you're using VSCode you can search throughout the whole codebase with the search feature. 

Here's the example he provided in the comments
```
validators={[ emptyValidator("grade") ]}
```